Core/GDCore/Project/ExtensionPropertiesManager.h

```diff
@@ -0,0 +1,55 @@
+/*
```

Comment:
Did you mean something like this? Or did you mean something else?

Reply:
I am awaiting confirmation @4ian (I don't want to do something wrong again)

Reply:
I see that you're still storing an `std::shared_ptr<gd::PlatformExtension> extension` and `std::map<gd::String, gd::PropertyDescriptor> extraData` in your "ExtensionProperty", why is that?
I think that you're seeing things as more complex than they should be.
Don't store "metadata" - they are in the extension and you don't need a pointer to them in the project. Just store:
- the name
- what is its value.

Basically it's just a map :) With a function to get a value, and another one to set a value (given as parameter the property name).
With SerializeTo/UnserializeFrom methods, and you're done.

Then in the IDE, you use `PropertiesEditor` component to show the properties of each extension. You pass it the properties coming from the extension, as I explained.
You need to pass a function that will be called when a property value is modified by the user, and another one used to read the current value. In this case, you use `project.getExtensionPropertiesManager().getValue(extension, propertyName)` or ``project.getExtensionPropertiesManager().setValue(extension, propertyName, newValue)`.

Reply:
Ohhh I get it. I thought about something way too complicated indeed.